services/QuillLMS/lib/rule_feedback_history.rb

```diff
@@ -26,6 +26,31 @@ def self.exec_query(conjunction:, activity_id:)
         rules = Comprehension::Rule.find_by_sql(sql)
     end
 
+    def self.feedback_history_to_json(f_h)
+        {
+            response_id: f_h.id,
+            datetime: f_h.updated_at,
+            entry: f_h.entry,
+            highlight: f_h.metadata.class == Hash ? f_h.metadata['highlight'] : '',
+            view_session_url: 'Not yet available',
+            strength: f_h.feedback_history_ratings.order(updated_at: :desc).first&.rating
+        }
+    end
+
+    def self.generate_rulewise_report(rule_uid)
+        feedback_histories = FeedbackHistory.where(rule_uid: rule_uid)
+        response_jsons = []
+        feedback_histories.each do |f_h|
+            response_jsons.append(feedback_history_to_json(f_h))
+        end
```

Comment:
Is this more cleanly understood as a map?
```ruby
response_jsons = feedback_histories.map { |f_h| feedback_history_to_json(f_h) }
```
Feels slightly cleaner than creating an empty array and then populating it.

Reply:
Absolutely; I must have been tired. I just made a patch which I will apply in version 2 later this week